worker/stateconverter/converter_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package stateconverter_test
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/juju/errors"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/model"
+	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/rpc/params"
+	"github.com/juju/juju/worker/stateconverter"
+	"github.com/juju/juju/worker/stateconverter/mocks"
+)
+
+var _ = gc.Suite(&converterSuite{})
+
+type converterSuite struct {
+	machine  *mocks.MockMachine
+	machiner *mocks.MockMachiner
+}
+
+func (s *converterSuite) TestSetUp(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	s.machiner.EXPECT().Machine(gomock.Any()).Return(s.machine, nil)
+	s.machine.EXPECT().Watch().Return(nil, nil)
+
+	conv := s.newConverter()
+	_, err := conv.SetUp()
+	c.Assert(err, jc.ErrorIsNil)
+}
+
+func (s *converterSuite) TestSetupMachinerErr(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	expectedError := errors.NotValidf("machine tag")
+	s.machiner.EXPECT().Machine(gomock.Any()).Return(nil, expectedError)
+
+	conv := s.newConverter()
+	w, err := conv.SetUp()
+	c.Assert(err, jc.Satisfies, errors.IsNotValid)
+	c.Assert(w, gc.IsNil)
+}
+
+func (s *converterSuite) TestSetupWatchErr(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	s.machiner.EXPECT().Machine(gomock.Any()).Return(s.machine, nil)
+	expectedError := errors.NotValidf("machine tag")
+	s.machine.EXPECT().Watch().Return(nil, expectedError)
+
+	conv := s.newConverter()
+	w, err := conv.SetUp()
+	c.Assert(err, jc.Satisfies, errors.IsNotValid)
+	c.Assert(w, gc.IsNil)
+}
+
+func (s *converterSuite) TestHandle(c *gc.C) {
+	defer s.setupMocks(c).Finish()
+	s.machiner.EXPECT().Machine(gomock.Any()).Return(s.machine, nil)
+	s.machine.EXPECT().Watch().Return(nil, nil)
+	jobs := params.JobsResult{Jobs: []model.MachineJob{model.JobHostUnits, model.JobManageModel}}
+	s.machine.EXPECT().Jobs().Return(&jobs, nil)
+
+	conv := s.newConverter()
+	_, err := conv.SetUp()
+	c.Assert(err, gc.IsNil)
+	err = conv.Handle(nil)
+	// Since machine has model.JobManageModel, we expect an error
+	// which will get machineTag to restart.
+	c.Assert(err.Error(), gc.Equals, "bounce agent to pick up new jobs")
+}
+
+func (s *converterSuite) TestHandleNoManageEnviron(c *gc.C) {
```

Comment:
would be good to avoid the term "ManageEnviron".
why not "TestHandleNotController" or something?